### PR TITLE
ci: migrate to github actions

### PR DESCRIPTION
**Pull request by maniax89**
#### What do these changes do/fix?

Since travis is limited to 10,000 credits of build time for open source projects, migrating to github actions instead for our same workflow.

#### How do you test/verify these changes?

Need to validate this still works on both `master` and `release/*` branches

### CODE REVIEW

**Review by kabeaty — APPROVED** [review]

**Review by NickMele — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  CI: true
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+           # pulls all commits (needed for lerna to correctly version)
```
Comment (jhpedemonte) [inline .github/workflows/ci.yaml:13]: Thanks for these comments.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          fetch-depth: "0"
+      # pulls all tags (needed for lerna to correctly version)
+      - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+      - name: Setup Node ${{ matrix.node_version }}
```
Comment (jhpedemonte) [inline .github/workflows/ci.yaml:17]: Where does `matrix.node_version` come from?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12
+      - name: Install
+        run: yarn install
+      - name: Lint
+        run: yarn lint
+      - name: Build
+        run: npx lerna run build
```
Comment (jhpedemonte) [inline .github/workflows/ci.yaml:26]: Why use `npx`? Isn't `lerna` installed?

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Build storybook
+        run: yarn workspace @ibm-watson/discovery-react-components run storybook:build
```
Comment (jhpedemonte) [inline .github/workflows/ci.yaml:28]: Any reason to not just do `lerna run storybook:build`, so we keep all "main" commands at the top level?

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Test
+        run: npx lerna run test
+      - name: NPM Identity
+        if: github.ref == 'refs/heads/master' || contains(github.ref, 'release/')
+        run: |
+          echo "registry=//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
+          npm whoami
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (jhpedemonte) [inline .github/workflows/ci.yaml:37]: Where are secrets defined for GH actions? Is it currently set up for your NPM token?

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+          git config --global user.name "Watson Github Bot"
+          git config remote.origin.url https://$GH_TOKEN@github.com/$GITHUB_REPOSITORY
+        env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+      - name: Publish Beta (Master)
+        if: github.ref == 'refs/heads/master'
+        run: npx lerna publish -y --conventional-prerelease --no-changelog --preid beta --dist-tag beta
+      - name: Publish Release Candidate (Release)
+        if: contains(github.ref, 'release/')
+        run: npx lerna publish -y --conventional-prerelease --no-changelog --preid rc --dist-tag rc
```
Comment (jhpedemonte) [inline .github/workflows/ci.yaml:52]: No notifications?

Comment (maniax89) [reply .github/workflows/ci.yaml:61]: we already have the slack integration setup in `#disco-widgets-deploy` - can you think of what other kinds of notifications you want? like when `master` branch breaks?

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        with:
+           # pulls all commits (needed for lerna to correctly version)
+          fetch-depth: "0"
+      # pulls all tags (needed for lerna to correctly version)
+      - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: '**/node_modules'
```
Comment (maniax89) [inline .github/workflows/ci.yaml:21]: not sure if this actually works yet

Comment (maniax89) [reply .github/workflows/ci.yaml]: confirmed this works

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (jhpedemonte) [reply .github/workflows/ci.yaml]: Can you add comments pointing to docs, like this link? Just makes it easier to click from within the file when needing to.

**Review by jhpedemonte — APPROVED** [review]

**Pull request merged**